Tweaks

smolkaj opened the pull request:
- Add .bazel suffix to WORKSPACE/BUILD files to avoid confusing people unfamiliar with Bazel.
- Make p4runtime dependency more robust:
  - Use same version as cmake, i.e. the submodule loaded as `control-plane/p4runtime`.
  - Don't use custom BUILD file.
     - This required first fixing the upstream Bazel rules... (see https://github.com/p4lang/p4-constraints/pull/27)
     - ... as well as updating the submodule to point to the latest commit.
- Make googletest dependency more robust: 
  - Use same version as cmake, i.e. the submodule loaded as `test/frameworks/gtest`.
  - Don't use custom BUILD file.
    - This required updating the submodule to a newer upstream version of gtest.
- Factor repository setup into `p4c_deps` rule to make p4c easy to use by other Bazel projects.

smolkaj committed:
Use .bazel extension for Bazel files.

Using WORSKPACE and BUILD without extensions is not friendly to non Bazel users.

smolkaj committed:
Use .gitsubmodule-version of p4runtime in Bazel rule.

Otherwise, we would have to keep the two versions of p4runtime in sync.

Caveat: This required fixing the p4runtime Bazel rules upstream, see
https://github.com/p4lang/p4runtime/pull/297. These fixes have not been merged
yet, so this commit pins the p4runtime submodule to a commit that's hasn't been
merged to master yet. Fix once the p4runtime PR has been merged.

smolkaj committed:
Attempt to factor all dependencies into seperate file, to make p4c ueasier to use from other Bazel projects.

smolkaj committed:
Try to simplify dependencies.

smolkaj committed:
Use same gtest as cmake; upgrade gtest to make that possible.

smolkaj committed:
Clean up BUILD file

smolkaj commented:
@EileenFeng @stefanheule could you take a look if this looks reasonable?

smolkaj committed:
Update .gitmodules

stefanheule commented on BUILD.bazel:
```diff
@@ -1,14 +1,11 @@
-
 P4C_BUILD_DEFAULT_COPTS = [
     "-DCONFIG_PKGDATADIR=\\\"external/p4lang_p4c\\\"",
     # This is a bit of a hack, but will work if the binary is executed by Bazel
```
I don't really understand this, is this hack necessary? (I know this is not something you changed, just curious).

smolkaj replied:
I'd like to understand that as well. Not sure yet.

smolkaj replied:
Let's leave this for another time.

stefanheule commented on .gitignore:
```diff
@@ -37,3 +37,6 @@ build/
 
 # macOS files
```
Are we still working on a p4_library bazel function?

smolkaj replied:
Yes.

smolkaj replied:
Done.

stefanheule commented:
Overall this looks great though, I'm super excited to get p4c opensource bazel rules.

smolkaj committed:
Refactor p4c Bazel rule and make sure it works using CI and example project.

* Use local repositories for local build, but fall back to git repositories for
  3rd party build. This is necessary due to
  https://github.com/bazelbuild/bazel/issues/11573

* Rename lex -> flex.

* Ignore bazel sub-projects so `bazel build //...` works.

smolkaj committed:
tweaks

smolkaj committed:
Stop using local_repository altogether; this should be less confusing if things break.

smolkaj committed:
Merge pull request #1 from smolkaj/bazel2

Refactor p4c Bazel rule and make sure it works using CI and example project.

smolkaj committed:
No need to update git submodules anymore.
